### ISSUE TITLE
Implement silhouette_score as quality metric

### ISSUE THREAD

**Issue by jsiegle** [AllenInstitute/ecephys_spike_sorting#41]
See: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.silhouette_score.html

This is a standard metric of cluster quality, suggested by Jai Bhagat
